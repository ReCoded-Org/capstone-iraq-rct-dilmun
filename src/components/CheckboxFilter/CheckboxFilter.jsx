import React from 'react'
import { PropTypes } from 'prop-types'
import { useTranslation } from 'react-i18next'

export default function CheckboxFilter({ChangeCheckbox}) {
    const { t } = useTranslation()
    const catagories = t('additem.cat', { returnObjects: true })


    const result = Object.keys(catagories).map(key => catagories[key])
    return (
        <div onChange={ChangeCheckbox} className="flex flex-col">
            <p className=" text-lg font-medium mb-3">Category</p>
            { result.map(category => {
                return (
                    <label htmlFor={category.value} className="inline-flex items-center mt-3">
                        <input type="checkbox" id={category.value} className="h-5 w-5"/><span className="ml-2">{category.value}</span>
                    </label>
                )
            })}
            <hr className="my-4" />
        </div>
    )
}

CheckboxFilter.propTypes = {
    ChangeCheckbox: PropTypes.func
}

CheckboxFilter.defaultProps = {
    ChangeCheckbox: () => {}
}