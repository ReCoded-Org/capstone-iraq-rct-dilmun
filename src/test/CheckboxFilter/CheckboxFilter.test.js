import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import store from '../../redux/store'
import CheckboxFilter from '../../components/CheckboxFilter'

it('Test: CheckboxFilter Passed the test', () => {
  const tree = renderer
    .create(
      <Router>
        <Provider store={store}>
          <CheckboxFilter />
        </Provider>
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
