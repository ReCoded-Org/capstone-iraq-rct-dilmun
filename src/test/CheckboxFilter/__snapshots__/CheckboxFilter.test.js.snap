// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: CheckboxFilter Passed the test 1`] = `
<div
  className="flex flex-col"
  onChange={[Function]}
>
  <p
    className=" text-lg font-medium mb-3"
  >
    Category
  </p>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <label
    className="inline-flex items-center mt-3"
  >
    <input
      checked={false}
      className="h-5 w-5"
      type="checkbox"
    />
    <span
      className="ml-2"
    />
  </label>
  <hr
    className="my-4"
  />
</div>
`;
